src/Psalm/Internal/Type/Comparator/GenericTypeComparator.php

```diff
@@ -136,36 +136,33 @@ public static function isContainedBy(
                 && !$container_type_part instanceof TIterable
                 && !$container_param->hasTemplate()
                 && !$input_param->hasTemplate()
+                && !($container_type_params_covariant[$i] ?? false)
```

Comment:
It's here because I think there is no sense do specialization for covariant parameter.